### PR TITLE
Feature/event pictures

### PR DESCRIPTION
**Pull request by ArtemisSoftware**

### CODE REVIEW

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/composables/PhotoGallery.kt [hunk 1]
```diff
@@ -22,183 +29,186 @@ import com.artemissoftware.core.presentation.theme.Light2
...
+                borderWidth = 2.dp,
+                iconColor = LightBlue,
+                borderColor = LightBlue,
+                padding = 8.dp,
+                size = 60.dp,
+                modifier = modifier,
+            )
+        }
+    }
+}
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/composables/PhotoGallery.kt:205]: You could just:

val location = if (picture is Picture.Local) picture.uri else picture.uri

And just define the TaskySquareIcon once

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailScreen.kt [hunk 2]
```diff
@@ -45,6 +51,28 @@ private fun EventDetailScreenContent(
...
+                        val size: Int = inputStream.available() ?: 0
+                        inputStream.close()
+
+                        if (size < ONE_MEGA_BYTE) {
+                            events(DetailEvents.AddPicture(uri = uri))
+                        } else {
+                            events(DetailEvents.ShowImageSizeError)
+                        }
+                    }
+                }
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailScreen.kt:71]: I much rather have this logic on the viewmodel
UI shouldn't have any logic

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/models/Picture.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.artemissoftware.tasky.agenda.presentation.detail.models
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/models/Picture.kt:1]: Isn't the picture used on Domain models?

**Review by MKiperszmid — APPROVED** [review]

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/composables/PhotoGallery.kt (hunk 1 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/composables/PhotoGallery.kt:205]: I cant do it beacuse picture.uri is a Uri and picture.url is a string

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailScreen.kt (hunk 2 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailScreen.kt]: Refactored

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/models/Picture.kt (hunk 3 above)
Comment (MKiperszmid) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/models/Picture.kt:1]: Right. You need to extract the Android part to make it domain specific

But also, Event should have pictures in it. What object are you using to do it?

Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/models/Picture.kt:1]: I am still working on it

**Pull request merged**